catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/test/java/org/codice/ddf/catalog/subscriptionstore/common/CachedSubscriptionTest.java

```diff
@@ -0,0 +1,158 @@
+        when(mockBundle.getBundleId()).thenReturn(10L);
+
+        when(mockBundleContext.registerService(anyString(), anyObject(), anyObject())).thenReturn(
+                mockRegistration);
+
+        when(mockFactory.getTypeName()).thenReturn(TYPE);
+        when(mockFactory.createSubscription(anyObject())).thenReturn(mockSubscription);
+
+        metadata = new SubscriptionMetadata(TYPE, FILTER, CALLBACK);
+        cachedSubscription = new CachedSubscription(metadata) {
+            protected BundleContext getBundleContext() {
+                return mockBundleContext;
+            }
+        };
+    }
+
+    @Test(expected = SubscriptionRegistrationException.class)
+    public void testUnregisterThrowsException() {
+        cachedSubscription.unregisterSubscription();
+    }
+
+    @Test
+    public void testUnregister() {
+        cachedSubscription.registerSubscription(mockSubscription);
+        cachedSubscription.unregisterSubscription();
+        verify(mockRegistration).unregister();
+
+        // Need to ensure the exception happened on the 2nd call to unregister
+        boolean registrationWasSetToNull = false;
+        try {
+            cachedSubscription.unregisterSubscription();
+        } catch (SubscriptionRegistrationException e) {
+            registrationWasSetToNull = true;
+        }
+        assertThat(registrationWasSetToNull, is(true));
+    }
+
+    @Test(expected = SubscriptionRegistrationException.class)
+    public void testRegisterSubscriptionThrowsException() {
+        when(mockBundleContext.registerService(anyString(), anyObject(), anyObject())).thenReturn(
+                null);
+        cachedSubscription.registerSubscription(mockSubscription);
+    }
+
+    @Test
+    public void testRegisterSubscription() {
+        cachedSubscription.registerSubscription(mockSubscription);
+        validateRegistration();
+    }
+
+    @Test(expected = SubscriptionRegistrationException.class)
+    public void testRegisterWithFactoryThrowsException() {
+        when(mockFactory.getTypeName()).thenReturn(NOT_TYPE);
+        cachedSubscription.registerSubscription(mockFactory);
+    }
+
+    @Test(expected = SubscriptionRegistrationException.class)
+    public void testRegisterWithFactoryReturningNull() {
+        when(mockFactory.createSubscription(anyObject())).thenReturn(null);
+        cachedSubscription.registerSubscription(mockFactory);
+    }
+
+    @Test
+    public void testRegisterWithFactory() {
+        cachedSubscription.registerSubscription(mockFactory);
+        validateRegistration();
+    }
+
+    private void validateRegistration() {
+        ArgumentCaptor<Dictionary> argCaptor = ArgumentCaptor.forClass(Dictionary.class);
+        verify(mockBundleContext).registerService(eq(Subscription.class.getName()),
+                eq(mockSubscription),
+                argCaptor.capture());
+        Dictionary<String, String> props = argCaptor.getValue();
+        assertThat(props.size(), is(2));
```

Comment:
You should be able to call new `Map<String, String> props = new HashMap<>(argCaptor.getValue());`